[WIP]Update marginal likelihood for new VAE structure

Louisexu925 opened the pull request:
We updated credence’s structure where previous version have one latent space per time step vs the new structure’s latent space doesn’t depend on time t (we use one latent value to infer the whole time series). And thus we created a branch of credence_new_structure. Because of the structural change, many of the functions need to be updated, e.g. the marginal likelihood function. For this specific PR, it’s pretty much WIP and looking for initial feedback of the overall logic/math to the log likelihood calculation and thus no tests have added yet.

Louisexu925 committed:
Update marginal log likelihood

Louisexu925 committed:
Update marginal log likelihood

Louisexu925 committed:
Merge branch 'credence_new_structure_new_marginal_likelihood' of https://github.com/amzn/credence-to-causal-estimation into credence_new_structure_new_marginal_likelihood

Louisexu925 committed:
Resolve conflict

Louisexu925 committed:
Merge branch 'credence_new_structure_new_marginal_likelihood' of https://github.com/amzn/credence-to-causal-estimation into credence_new_structure_new_marginal_likelihood

Louisexu925 committed:
Bugfix: add kwargs that got deleted by accident

willettk commented:
Hi @Louisexu925 - a couple overall questions before looking at the details:

- Could you provide a little more detail on what this change is intended to do, and why it's necessary for REMI?
- Are there any tests you've run (or that can be run) to make sure that it works?
- Are there any changes to the version of `torch` required for the package?
- The PR is for the branch `credence_new_structure` - do you want to put these changes there, or ultimately into `mainline`?

Louisexu925 commented:
Close the PR because

Louisexu925 commented:
Could you provide a little more detail on what this change is intended to do, and why it's necessary for REMI? 
A: Please refer to the updated description of the PR  .

Are there any tests you've run (or that can be run) to make sure that it works?
A: Because the PR is pretty much PR and the goal is for initial logic review and thus no tests have added yet. 

Are there any changes to the version of torch required for the package?
A: No.

The PR is for the branch credence_new_structure - do you want to put these changes there, or ultimately into mainline?
A: Yes,  We are developing a new branch, credence_new_structure, for the new structure of Credence. And thus this PR will first merged with that branch. And once the credence_new_structure branch is ready, we will merge it to mainline.

Louisexu925 committed:
Remove sensitive information from the notebook

Will later update the notebook to use new public available dataset in  the notebook.

Louisexu925 commented:
Closing the PR because we no longer using marginal likelihood as the main validation metrics to validate synthetic data generated by Credence.
